File: java/src/questao_4/Main.java
package questao_4;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import questao_4.Person;

public class Main {
    public static int findPeopleWithMoreThanNinetyKg(List<Person> people) {
        int count = 0;
        for (Person p : people) {
            if (p.getWeight() > 90) {
                count++;
            }
        }
        return count;
    }

    public static double calculateAgeAverage(List<Person> people) {
        int sum = 0;
        for (Person p : people) {
            sum += p.getAge();
        }
        return ((double) sum) / people.size();
    }

    public static void main(String[] args) {
        List<Person> people = new ArrayList<>();

        Scanner scanner = new Scanner(System.in);
        while (people.size() < 7) {
            System.out.println("Digite a sua idade:");
            int age = scanner.nextInt();
            System.out.println("Digite o seu peso:");
            double weight = scanner.nextDouble();
            people.add(new Person(weight, age));
            System.out.println("\n----------------------");
            System.out.println("Informações salvas!");
            System.out.println("----------------------\n");
            System.out.println("Próximo!\n");
        }
        scanner.close();
        System.out.println("Quantidade de pessoas com mais de 90Kg: " + findPeopleWithMoreThanNinetyKg(people));
        System.out.println("A média de idades é: " + calculateAgeAverage(people));
    }
}

File: python/src/questao_4/main.py
from  person import Person
from typing import List

def findPeopleWithMoreThanNinetyKg(people:List[Person]):
    count = 0
    for p in people:
        if p.get_weight() > 90:
            count+=1
    
    return count


def calculateAgeAverage(people:List[Person]):
    sum = 0
    for p in people:
        sum+=p.get_age()
    
    return (sum/len(people))



def main():
    people = []

    while len(people) < 7:
        age = int(input("Digite a sua idade:"))
        weight = float(input("Digite o seu peso:"))
        people.append(Person(weight,age))
        print("\n")
        print("----------------------")
        print("Informações salvas!")
        print("----------------------")
        print("\n")
        print("Proximo!")
        print("")
    
    print("Quantida de pessoas com mais de 90Kg: ",findPeopleWithMoreThanNinetyKg(people))
    print("A media de idades é: ",calculateAgeAverage(people))




if __name__ == '__main__':
    main()